clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -2229,7 +2229,10 @@ class CIRGlobalOpLowering
     if (auto constArr =
             mlir::dyn_cast<mlir::cir::ConstArrayAttr>(init.value())) {
       if (auto attr = mlir::dyn_cast<mlir::StringAttr>(constArr.getElts())) {
-        init = rewriter.getStringAttr(attr.getValue());
+        llvm::SmallString<256> literal(attr.getValue());
+        if (constArr.getTrailingZerosNum())
```

Comment:
Hmm. For regular strings we just emit the string directly (https://godbolt.org/z/a6xWGx3hf), but for trailing NUL bytes we use trailing zeros (https://godbolt.org/z/r54vvKPb7). Maybe we shouldn't be generating the trailing zeroes instead (assuming there's no technical limitation around that)?

Reply:
LLVM will generate the trailing zero: https://godbolt.org/z/4G75e4P1o . I feel better if we can get align with LLVM in the traditional pipeline. (this is LowerToLLVM.cpp and it shouldn't affect how we generate cir)

Reply:
Perhaps OG should be fixed too? Anyways, in the meantime I'm sold on keeping it consistent with OG